auth-internal/src/main/java/com/sequenceiq/cloudbreak/auth/security/internal/InternalCrnModifier.java

```diff
@@ -63,12 +63,7 @@ public Object changeInternalCrn(ProceedingJoinPoint proceedingJoinPoint) {
 
     private String getAccountIdModifiedCrn(String userCrnString, String accountId) {
         Crn userCrn = Crn.fromString(userCrnString);
-        Crn newUserCrn = Crn.builder()
-                .setService(userCrn.getService())
-                .setAccountId(accountId)
-                .setResourceType(userCrn.getResourceType())
-                .setResource(userCrn.getResource())
-                .build();
+        Crn newUserCrn = Crn.copyWithDifferentAccountId(userCrn, accountId);
```

Comment:
looks dangerous :)

Reply:
what do you mean by that? it does the same as the current code

Reply:
Yes, I know, but that logic was private to that InternalCrnModifier implementation, while the static Crn is used widely in the code. I'm just contemplating wether it's worth to expose that.

Reply:
setAccountId in Crn.builder() was not private, I mean right now everyone can change accountId If he/she want to do so. one solution can be to make the method package private or something like that but InternalCrnModifier and Crn placed in different module. We can move one of them but it can cause other issues.